lsp-mode.el

```diff
@@ -7563,13 +7570,15 @@ The server(s) will be started in the buffer when it has finished."
        ((setq clients (unless matching-clients
                         (lsp--filter-clients (-andfn #'lsp--matching-clients?
                                                      (-not #'lsp--server-binary-present?)))))
-        (lsp--warn "The following servers support current file but do not have automatic installation configuration: %s
+        (if lsp-server-allow-automatic-install
+            (lsp--warn "The following servers support current file but do not have automatic installation configuration: %s
 You may find the installation instructions at https://github.com/emacs-lsp/lsp-mode/#supported-languages.
 (If you have already installed the server check *lsp-log*)."
-                   (mapconcat (lambda (client)
-                                (symbol-name (lsp--client-server-id client)))
-                              clients
-                              " ")))
+                       (mapconcat (lambda (client)
+                                    (symbol-name (lsp--client-server-id client)))
+                                  clients
+                                  " "))
+          (lsp--warn "LSP server is not present and automatic install of LSP servers has been disabled. Check the variable lsp-server-allow-automatic-install.")))
```

Comment:
Perhaps this should be "The following servers could be installed but automatic installation was disabled: %s" matching-clients

Reply:
> private lsp-mode variables,

FWIW lsp--client-* should be made public to allow further client customizations.